src/rpc/blockchain.cpp

```diff
@@ -491,19 +491,21 @@ static void entryToJSON(const CTxMemPool& pool, UniValue& info, const CTxMemPool
     fees.pushKV("ancestor", ValueFromAmount(e.GetModFeesWithAncestors()));
     fees.pushKV("descendant", ValueFromAmount(e.GetModFeesWithDescendants()));
     info.pushKV("fees", fees);
-
+    // TODO: top level fee fields are deprecated. this method should be refactored in v24 to completely remove them
+    if (IsDeprecatedRPCEnabled("fees")) {
+        info.pushKV("fee", ValueFromAmount(e.GetFee()));
+        info.pushKV("modifiedfee", ValueFromAmount(e.GetModifiedFee()));
+        info.pushKV("descendantfees", e.GetModFeesWithDescendants());
+        info.pushKV("ancestorfees", e.GetModFeesWithAncestors());
+    }
```

Comment:
Might be more user-friendly to not alter the field order here so that it is the same as the help. E.g. insert the ISDeprecated... conditional (or cache the result of the call to a local bool variable) before the fields where they already are.

Reply:
The code will go away soon, so I don't think this gives much benefit.

Reply:
i noticed that the current order doesn't match the help (fees object is listed after wtxid but is first in the returned object), so didn't worry about re-arranging fields, but i agree that it can be confusing for the user. 

i think grouping the fields makes it easier to deprecate them later, so maybe a better solution would be to re-arrange the help to match the actual ordering? or just leave it as is per @MarcoFalke 's comment, i dont have a strong opinion either way

Reply:
I would say, please don't re-arrange the help to make up for something the code can easily do.

Yes, I agree it's not of earth-shattering importance, but I also think it's a more important aspect than in which file to put the deprecation test or which way to check for deprecation. It affects users.

Reply:
> The code will go away soon, so I don't think this gives much benefit.

Same for the deprecation test and which way to check for deprecation, but the pull was updated for these. I guess it's a question of degree of focus on users. Either way, not a big deal.

Reply:
> I guess it's a question of degree of focus on users

good point, @jonatack , i'll update based on your suggestion

Reply:
Happy to re-review ASAP if you do (it's not a blocker).

Reply:
> i noticed that the current order doesn't match the help (fees object is listed after wtxid but is first in the returned object)

Oh, I see now what you meant. Hm, the fees output was done in 7de1de7da49 in 2018 and the fees help in fa6b061fc11 in 2020. It looks like the later commit auto-moved JSON objects after top-level fields? So your change ~seems correct, unless @glozow disagrees whether better to change the help to be like the output.

Reply:
> i noticed that the current order doesn't match the help (fees object is listed after wtxid but is first in the returned object)

Oh, I see now what you meant. Hm, the fees output/help mismatch was done in 7de1de7da49. ISTM we usually place JSON objects after top-level fields? So your change ~seems correct, unless other reviewers think it's better to change the help to be like the output.

Reply:
> JSON objects after top-level fields

that's the convention i'm used to (and much prefer)